protocol.go

```diff
@@ -0,0 +1,152 @@
+package proxyproto
```

Comment:
Just realized I forgot to add any copyright notice, will at least add that it's derived from https://github.com/armon/go-proxyproto/blob/master/protocol.go